packages/core/src/internal/proc.js

```diff
@@ -103,12 +108,13 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
         }
         mainTask.cont(result.value)
       }
-    } catch (error) {
+    } catch (err) {
       if (mainTask.status === CANCELLED) {
-        throw error
+        throw err
       }
       mainTask.status = ABORTED
-      mainTask.cont(error, true)
+
+      mainTask.cont(isErr ? arg : new SagaError(err), true)
```

Comment:
this changes how it behave previously - could u explain how it behave and how it will behave after this change?

previously locally caught error was passed to `mainTask.cont`, currently `arg` gets reused

Reply:
ok, after a thought i think i know how it works - previously arg and locally caught error were actually the same, right?

Reply:
yes, I must add the comment about it. 
right now there are 2 source of errors for `next`
1 an exception passed as an argument. we will throw it into iterator and will catch it again. to remain stack trace we should reuse `arg`.
2 an exception raised while resuming iterator, so we wrap it